addons/account/models/chart_template.py

```diff
@@ -295,18 +295,19 @@ def tax_template_changed(tax, template):
                 elif model_name == 'account.account':
                     # Point or create xmlid to existing record to avoid duplicate code
                     account = self.ref(xmlid, raise_if_not_found=False)
-                    if not account or (account and account.code != values['code']):
-                        existing_account = self.env['account.account'].search([
-                            ('code', '=', values['code']),
-                            ('company_id', '=', company.id),
-                        ])
-                        if existing_account:
+                    normalized_code = f'{values["code"]:<0{int(template_data.get("code_digits", 6))}}'
+                    if not account or not re.match(f'^{values["code"]}0*$', account.code):
+                        query = self.env['account.account']._search([('company_id', '=', company.id)])
+                        query.add_where("account_account.code SIMILAR TO %s", [f'{values["code"]}0*'])
+                        accounts = self.env['account.account'].browse(query)
+                        account = accounts.sorted(key=lambda x: x.code != normalized_code)[0] if accounts else None
```

Comment:
Why are you making the sort here? Can't you just take the first one? (not even sure it is valid to have `123` and `123000` for instance :thinking: (even if there is no hard constraint)

Reply:
I thought it more appropriate to give priority the one with the completely matching code if it comes to that. You never know what you might find :man_shrugging: